server/server.cc

```diff
@@ -0,0 +1,129 @@
+#include "server.hh"
+
+using namespace seastar;
+
+static uint16_t calc_hash(const std::string& s) {
+    uint16_t res = 0;
+    for (char c: s) {
+        res += (uint8_t) c;
+    }
+    return res % smp::count;
+}
+
+std::unordered_map<std::string, std::string> tcp_server::rust_data =
+    std::unordered_map<std::string, std::string>();
+
+future<> tcp_server::listen(ipv4_addr addr) {
+    listen_options lo;
+    lo.proto = transport::TCP;
+    lo.reuse_address = true;
+    _tcp_listener = seastar::listen(make_ipv4_address(addr), lo);
+    do_accept(_tcp_listener);
+    co_return;
+}
+
+future<> tcp_server::stop() {
+    co_return;
+}
+
+void tcp_server::do_accept(server_socket& listener) {
+    (void)listener.accept().then([this, &listener] (accept_result ar) mutable {
+        connected_socket fd = std::move(ar.connection);
+        socket_address addr = std::move(ar.remote_address);
+        auto conn = new connection(*this, std::move(fd), addr);
```

Comment:
Let's not use raw pointers if unnecessary. It's easy to forget to delete them. Instead, please use `std::unique_ptr`/`seastar::lw_shared_ptr` or wrap the `conn->process()` in a `seastar::do_with` invocation which constructs the `connection` object.